### PR TITLE
Making sure inlineFrags array contains ONLY fragments.

### PR DESCRIPTION
**Pull request by seanchoi-dev**
Found a bug that if a link that is not fragment contains `#_inline` in its URL, it will be treated like non-decorated fragment.

Fixed result will show just a regular anchor link if it is not a fragment URL.

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/seanchoi/inline-test?martech=off
- After: https://inline-tag-bug--milo--adobecom.hlx.page/drafts/seanchoi/inline-test?martech=off

- Production page that is using fragments with `#_inline` (making sure this fix doesn't break anything):
https://main--homepage--adobecom.hlx.page/homepage/index-loggedout?milolibs=inline-tag-bug&martech=off

### CODE REVIEW

File: libs/utils/utils.js
Comment (chrischrischris) [inline libs/utils/utils.js:995]:
```suggestion
  const inlineFrags = [...section.el.querySelectorAll('a[href*="#_inline"][class~="fragment"]')];
```

**Review by mokimo — COMMENTED** [review]
Is that really a bug or a feature? Did you encounter `#_inline` anywhere on production that wasn't meant to be a fragment?

It's quite the specific hash - but @chrischrischris might have more input. I'm not sure if we ever want fragments living potentially anywhere.

**Review by mokimo — APPROVED** [review]

**Comment by mokimo** [comment]
Also 2 more comments, preferably before merging:
I'd love to have a QA verify the changes & create and link a Ticket.
Those `#_inline` fragments are already being used and this change potentially affects production pages

File: libs/utils/utils.js
Comment (auniverseaway) [inline libs/utils/utils.js:999]: This seems like it would be better and simpler...
```suggestion
  const inlineFrags = [...section.el.querySelectorAll('a.fragment[href*="#_inline"]')];
```

**Review by auniverseaway — CHANGES_REQUESTED** [review]
Please fact check my recommendation, but I think it should work.

**Review by chrischrischris — APPROVED** [review]

**Review by sukamat — APPROVED** [review]

**Review by overmyheadandbody — APPROVED** [review]

**Comment by honstar** [comment]
@seanchoi-dev , what's the status on this PR? Looks like you have all the required approvals, please check and merge or close it if this is no more relevant.

**Comment by mokimo** [comment]
@seanchoi-dev feel free to reopen if this is still relevant

**Pull request closed without merging**
